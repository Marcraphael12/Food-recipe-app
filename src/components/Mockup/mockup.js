import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';

import './mockup.scss';

// import fetchCategories from '../../redux/API/features/async_thunk';
// import { WorkDone } from '../Home/Home';

const Mockup = () => {
  const data = useSelector((state) => state.food.food);
  const content = Object.values(data);
  return (
    <div className="mockup">
      <NavLink className="back" to="/">â—€</NavLink>
      {content.filter((food) => food.joined).map((food) => (
        <div key={food.category_id}>
          <img src={food.category_image} alt={food.category_name} />
          <h2>{food.category_name}</h2>
          <p>{food.description}</p>
          <NavLink to="/"><button type="button">see all foods</button></NavLink>
        </div>
      ))}
    </div>
  );
};

export default Mockup;
