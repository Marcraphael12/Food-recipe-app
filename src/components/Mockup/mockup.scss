.mockup {
  width: 100%;
  max-width: 600px;
  height: auto;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-top: 20px;

  >.back {
    width: 50px;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #002364;
    border-radius: 10px;
    box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1);
    padding: 20px;
    margin-bottom: 20px;
  }

  >div {
    width: 100%;
    height: auto;
    background-color: #002364;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    padding: 10px;
    margin-top: 10px;
    border: 2px solid #0d43aa;
    border-radius: 15px;

    >img {
      width: 100%;
      height: auto;
    }

    >h2 {
      font-family: "Montserrat", sans-serif;
      font-size: 16px;
      color: #fff;
      padding: 10px 0;
    }

    >p {
      width: 100%;
      height: auto;
      display: flex;
      justify-content: center;
      align-items: center;
      padding: 10px 0;
      text-align: center;
      font-size: 14px;
      color: #fff;
    }

    >a {
      width: 100%;
      height: auto;

      >button {
        width: 100%;
        height: auto;
        display: flex;
        justify-content: center;
        align-items: center;
        padding: 10px 0;
        background-color: #003699;
        border: none;
        border-radius: 15px;
        font-size: 14px;
        color: #fff;
        font-weight: bold;

        &:hover {
          cursor: pointer;
          background-color: #38b6ff;
          opacity: 0.8;
        }
      }
    }
  }
}
