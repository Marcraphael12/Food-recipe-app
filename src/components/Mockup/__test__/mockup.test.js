import TestRenderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../../redux/configureStore';
import Mockup from '../mockup';

describe('Render the categories list', () => {
  it('Renders the selected category correctly as mockup', () => {
    const categories = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Mockup />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(categories).toMatchSnapshot();
  });
});
