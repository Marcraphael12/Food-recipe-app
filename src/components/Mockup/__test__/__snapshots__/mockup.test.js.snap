// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the categories list Renders the selected category correctly as mockup 1`] = `
<div
  className="mockup"
>
  <a
    aria-current="page"
    className="back active"
    href="/"
    onClick={[Function]}
  >
    â—€
  </a>
</div>
`;
